src/compiler/types.ts

```diff
@@ -2056,14 +2056,15 @@ namespace ts {
         SuperInstance               = 0x00000100,  // Instance 'super' reference
         SuperStatic                 = 0x00000200,  // Static 'super' reference
         ContextChecked              = 0x00000400,  // Contextual types have been assigned
-        LexicalArguments            = 0x00000800,
-        CaptureArguments            = 0x00001000,  // Lexical 'arguments' used in body (for async functions)
+        AsyncMethodWithSuper        = 0x00000800,
+        AsyncMethodWithSuperBinding = 0x00001000,

```

Comment:
I'll add a comment in `checker.ts` per https://github.com/Microsoft/TypeScript/pull/5870#discussion_r50786034